keps/sig-network/1880-service-subnet/README.md

```diff
@@ -0,0 +1,558 @@
+  // IPReserved means the IP has been reserved and can not be allocated to any Service.
+	IPReserved IPAddressState = "Reserved"
+	// IPFree means that IP has not been allocated neither persisted.
+	IPFree IPAddressState = "Free"
+)
+```
+
+The name of the IPAddress has to be unique, so we can use it to provide consistency and avoid duplicate allocations.
+However, Kubernetes objects doesn't allow names with `:`, used by IPv6 Addresses. In addition, [IPv6 addresses
+can have different text representations](https://tools.ietf.org/html/rfc4291).
+
+To guarantee the name uniqueness, we can use the decimal representation of the IP address, since IP can be a 32 bits or a 128 bits number. To facilitate this operations we introduce two new utils functions:
+
+```go
+// IPToDecimal converts an IP to a string with its decimal representation
+func IPToDecimal(ip net.IP) string {
+	if ip == nil {
+		return ""
+	}
+	i := new(big.Int)
+	// compare the address with its decimal representation
+	if ip.To4() != nil {
+		i.SetBytes(ip.To4())
+	} else if ip.To16() != nil {
+		i.SetBytes(ip.To16())
+	}
+	return i.String()
+}
+
+// DecimalToIP converts a string with the decimal representation of an IP address and returns the IP
+func DecimalToIP(ipDecimal string) net.IP {
+	if len(ipDecimal) == 0 {
+		return nil
+	}
+	// convert decimal representation to BigInt
+	i, ok := new(big.Int).SetString(ipDecimal, 10)
+	// if we can not convert the string to a number
+	// or the number has more than 16 bytes it is not an IP
+	if !ok || len(i.Bytes()) > 16 {
+		return nil
+	}
+	bufLen := net.IPv4len
+	if len(i.Bytes()) > net.IPv4len {
+		bufLen = net.IPv6len
+	}
+	// convert BigInt to IP address, truncate to IP Len bytes
+	r := append(make([]byte, bufLen), i.Bytes()...)
+	return net.IP(r[len(r)-bufLen:])
+}
+```
+
+
+```
+<<[UNRESOLVED IPAddress API Operations]>>
+
+* IPAddress objects created must reference the IPRange object, and validated against it. 
+The `address` field will contain the IP address representation, if no one is provided it will be defaulted useing the value obtained from the name. If the `address` is set it must match the `name`.
+
+* IPAddress status updates depend on the allocation model chosen, it may allowed to change after creation,
+
+QUESTION garbage collection, cluster and namespaced scoped objects. Which will be the best scope for these new objects? If we can make that IPAddresses are owned by the Services that allocate them, the garbage collector when deleting a Service will take care of the release????
+
+<<[/UNRESOLVED]>>
+```
+
+### NodePort
+
+TBD
+
+
+### Test Plan
+
+* Unit tests
+
+TBD
+
+* Integration Tests
+
+TBD
+
+* E2E tests 
+
+
+### Graduation Criteria
+
+The change must be transparent to the users, so there should not be any performance degradation
+or Services breaking changes. 
+
+### Upgrade / Downgrade Strategy
+
+The service-subnet and node-port range will keep using the flags for the initial configuration, it can
```

Comment:
We need more details here.  In what cases will the flags be used vs range objects?  Is it the case that as soon as one range exists (for a given family) then we start using that?  Is that only evaluated at apiserver startup?